### DISCUSSION TITLE
Unexpected end of JSON input

### DISCUSSION THREAD

**Discussion by kajal-7span** [rebing/graphql-laravel#1115] [Q&A]
### Versions:
- graphql-laravel Version: 8.6
- Laravel Version: 10.33.0
- PHP Version: 8.1

### Description:

I'm currently working on a GraphQL query that fetches a large dataset from my server. However, I'm encountering an "unexpected end of JSON input" error. This issue seems to be related to the size of the dataset. The query works fine with smaller datasets, but as the data grows, the error occurs. Basically the query is when I am exporting large number of products from my portal. Please check the attached image for reference.

![image](https://github.com/rebing/graphql-laravel/assets/107455736/328e1fd7-dbe6-42c6-baa1-315e74e85730)

To solve this issue I have already tried below options:

- Increased the size of memory limit
- Implemented whole export code using job and chunk.
- Restarted server and all artisan commands to clear cache and everything

Please suggest me the solution for it.

**Comment by illambo** [comment]
Hi, maybe pagination pattern can help you.

**Comment by kajal-7span** [comment]
How pagination pattern can work for me when I am Exporting or Importing the data @illambo

**Comment by illambo** [comment]
Maybe I not understand well, in your scenario the output of your query what is?
If the result are many items the pagination pattern can "chunk" and reduce size of output.

**Comment by mfn** [comment]
I've converted this to a discussion.

I see no evidence this is related to the library. In my experience, such cases are indeed connected to runtime corruption due to memory, timeout, etc.
